source/module_esolver/esolver_ks.cpp

```diff
@@ -623,6 +630,31 @@ void ESolver_KS<T, Device>::runner(const int istep, UnitCell& ucell)
                                  duration);
 #endif //__RAPIDJSON
 
+        // 12.5) rdmft, add by jghan 2024-04-08/2024-10-09
+        if ( PARAM.inp.ab_initio_type == "rdmft" )
+        {
+            // if ( (!GlobalC::exx_info.info_global.cal_exx && iter == 1) || one_step_exx )
+            if ( !GlobalC::exx_info.info_global.cal_exx || (GlobalC::exx_info.info_global.cal_exx && one_step_exx) )
+            {
+                ModuleBase::matrix occ_number_ks(this->pelec->wg);
+                for(int ik=0; ik < occ_number_ks.nr; ++ik)
+                {
+                    for(int inb=0; inb < occ_number_ks.nc; ++inb) occ_number_ks(ik, inb) /= this->kv.wk[ik];
+                }
+
+                this->update_elec_rdmft(occ_number_ks, *(this->psi));
+
+                //initialize the gradients of Etotal on wg and wfc, and set all elements to 0. 
+                ModuleBase::matrix E_gradient_occNum(this->pelec->wg.nr, this->pelec->wg.nc, true);
```

Comment:
would the name of variable be more precise if it were `dedfnk`, along with an annotation `// fnk: occupation of band n, at k` instead?